iceoryx_hoofs/utility/include/iox/detail/convert.inl

```diff
@@ -57,385 +58,370 @@ convert::toString(const Source& t) noexcept
-    {
-        return false;
-    }
-
-    auto call = IOX_POSIX_CALL(strtoull)(v, nullptr, STRTOULL_BASE).failureReturnValue(ULLONG_MAX).evaluate();
+    errno = 0;
+    char* end_ptr = nullptr;
 
-    if (call.has_error())
+    if (start_with_neg_sign(v))
     {
-        return false;
+        return iox::nullopt;
     }
 
-    if (call->value > std::numeric_limits<uint64_t>::max())
-    {
-        IOX_LOG(DEBUG, call->value << " too large, uint64_t overflow");
-        return false;
-    }
+    auto call = IOX_POSIX_CALL(strtoull)(v, &end_ptr, STRTOULL_BASE)
+                    .alwaysSuccess()
+                    .suppressErrorMessagesForErrnos(EINVAL, ERANGE)
+                    .evaluate();
 
-    dest = static_cast<uint64_t>(call->value);
-    return true;
+    return evaluate_return_value<uint64_t>(call, errno, end_ptr, v);
 }
 
 #ifdef __APPLE__
 /// introduced for mac os since unsigned long is not uint64_t despite it has the same size
 /// who knows why ¯\_(ツ)_/¯
 template <>
-inline bool convert::fromString<unsigned long>(const char* v, unsigned long& dest) noexcept
+inline iox::optional<unsigned long> convert::from_string<unsigned long>(const char* v) noexcept
 {
-    uint64_t temp{0};
-    bool retVal = fromString(v, temp);
-    dest = temp;
-    return retVal;
+    auto ret = from_string<uint64_t>(v);
+    if (!ret.has_value())
+    {
+        return iox::nullopt;
+    }
+    return iox::optional<unsigned long>{static_cast<unsigned long>(ret.value())};
 }
 #endif
 
 #if defined(__GNUC__) && (INTPTR_MAX == INT32_MAX)
 /// introduced for 32-bit arm-none-eabi-gcc since uintptr_t is not uint32_t despite it has the same size
 /// who knows why ¯\_(ツ)_/¯
 template <>
-inline bool convert::fromString<uintptr_t>(const char* v, uintptr_t& dest) noexcept
+inline iox::optional<uintptr_t> convert::from_string<uintptr_t>(const char* v) noexcept
 {
-    uint64_t temp{0};
-    bool retVal = fromString(v, temp);
-    dest = temp;
-    return retVal;
+    auto ret = from_string<uint32_t>(v);
+    if (!ret.has_value())
+    {
+        return iox::nullopt;
+    }
+    return iox::optional<uintptr_t>{static_cast<uintptr_t>(ret.value())};
 }
 #endif
 
 template <>
-inline bool convert::fromString<uint32_t>(const char* v, uint32_t& dest) noexcept
+inline iox::optional<uint32_t> convert::from_string<uint32_t>(const char* v) noexcept
 {
-    if (!stringIsNumberWithErrorMessage(v, NumberType::UNSIGNED_INTEGER))
+    errno = 0;
+    char* end_ptr = nullptr;
+
+    if (start_with_neg_sign(v))
     {
-        return false;
+        return iox::nullopt;
     }
 
-    auto call = IOX_POSIX_CALL(strtoull)(v, nullptr, STRTOULL_BASE).failureReturnValue(ULLONG_MAX).evaluate();
+    auto call = IOX_POSIX_CALL(strtoull)(v, &end_ptr, STRTOULL_BASE)
+                    .failureReturnValue(ULLONG_MAX)
+                    .suppressErrorMessagesForErrnos(EINVAL, ERANGE)
+                    .evaluate();
```

Comment:
If I understand correctly, this approach might fail in boundary cases because `evaluate_return_value` checks for errors first (using `.has_error()` of the return value), and using `failureReturnValue` can mislead the result in boundary conditions.

Therefore, if you intend to use it, you should switch to `alwaysSuccess()`. Additionally, `is_within_range` would become ineffective in this scenario, meaning it won't work for:
- `long long`
- `long`
- `int`

Please correct me if anything I miss.

Reply:
See [3afbea9](https://github.com/eclipse-iceoryx/iceoryx/pull/2135/commits/3afbea98e9f38108f7bd841d1985de24ced4fe67).

Reply:
I think with the changes you did it should just work :)
If it wouldn't work you would not have been able to remove my workaround for the relative pointer.
There is only one change which might be required to prevent warnings in `is_within_range`. There could be an `if constexpr` block to skip the check if the target type is equal to the source type.
It seems that the current compiler generation does not issue such a warning else we would have already noticed it for `unsigned long long`.